app/src/main/java/org/kiwix/kiwixmobile/nav/destination/library/OnlineLibraryFragment.kt

```diff
@@ -433,19 +433,22 @@ class OnlineLibraryFragment : BaseFragment(), FragmentActivityExtensions {
           })
           return
         }
-        else -> availableSpaceCalculator.hasAvailableSpaceFor(
-          item,
-          { downloadFile(item.book) },
-          {
-            libraryList.snack(
-              getString(R.string.download_no_space) +
-                "\n" + getString(R.string.space_available) + " " +
-                it,
-              R.string.download_change_storage,
-              ::showStorageSelectDialog
-            )
-          }
-        )
+        else -> if (sharedPreferenceUtil.showStorageOption) {
+          showStorageConfigureDialog()
```

Comment:
hi @cdhiraj40 ,

So now we have two version of build specifically for android 11 and above
a) Play-Store version:- App specific without ```MANAGE_EXTERNAL_PERMISSION```
b) F-Droid version:- App specific with ```MANAGE_EXTERNAL_PERMISSION```

So you have place https://github.com/kiwix/kiwix-android/blob/52f7002ba65f080ac2330b0d4b018ca73afe7d13/core/src/main/java/org/kiwix/kiwixmobile/core/utils/SharedPreferenceUtil.kt#L74 and ```Build.VERSION.SDK_INT >= Build.VERSION_CODES.R``` check before showing dialog.

Reply:
ah got it, okay will do!